src/system.h

```diff
@@ -643,6 +653,21 @@ extern UInt SyTimeChildrenSys ( void );
 
 /****************************************************************************
 **
+*F  CharHexDigit( <ch> ) . . . . . . . . . turn a single hex digit into Char 
+**
+*/
+static inline Char CharHexDigit( const Char ch ) {
+    if (ch >= 'a') {
+        return (ch - 'a' + 10);
+    } else if (ch >= 'A') {
+        return (ch - 'A' + 10);
+    } else {
+        return (ch - '0');
+    }
+};
```

Comment:
This is only used in one place... so unless we forsee use for it, why not just move it to `scanner.c` ? If something else ever needs it, we can still move it to a header (and then contemplate whether it needs extra safe guards or not etc.)